### ISSUE TITLE
unused-import false negative when typing.Annotated has str parameters

### ISSUE THREAD

**Issue by lggruspe** [PyCQA/pylint#7547]
### Bug description

example.py:

```python
# pylint: disable=missing-docstring
from pathlib import Path
import typing as t

example: t.Annotated[str, "Path"] = "/foo/bar"
```

### Command used

```shell
pylint example.py
```

### Pylint output

```shell
Success: no issues found in 1 source file

--------------------------------------------------------------------
Your code has been rated at 10.00/10 (previous run: 10.00/10, +0.00)
```

### Expected behavior

It should report `unused-import`, since the parameter to `Annotated` does not refer to `pathlib.Path`.

### Pylint version

```shell
pylint 2.15.3
astroid 2.12.10
Python 3.10.7 (main, Sep  7 2022, 00:00:00) [GCC 12.2.1 20220819 (Red Hat 12.2.1-1)]
```

### OS / Environment

fedora 36

**Comment by Pierre-Sassoulas** [comment]
I don't understand the example, what does the parameter of annotated refer to according to you ? I think ``"Path"`` does refer to ``pathlib.Path`` ?

**Comment by lggruspe** [comment]
What comes after the first parameter to `Annotated` is just ["an arbitrary list of Python values"](https://peps.python.org/pep-0593/#syntax).
Its interpretation is left to the library or tool that encounters the `Annotated` type: https://peps.python.org/pep-0593/#consuming-annotations.

So if I understand correctly, the `"Path"` in the example is just a string literal rather than a reference to a type.

**Comment by Pierre-Sassoulas** [comment]
Ok, I think you're right and it's a false negative. ``autoflake`` is not removing the import either, you could open the same issue in their repository if you want.

**Comment by DanielNoord** [comment]
Just to be sure, we should just ignore any second parameter to `Annotated` right? Or is there any case where it should be counted towards "usage".

**Comment by lggruspe** [comment]
I think so. mypy ignores the second parameter as well.
